packages/backend-app-api/src/services/implementations/auth/plugin/keys/StaticConfigPluginKeySource.ts

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Config } from '@backstage/config';
+import { HumanDuration, durationToMilliseconds } from '@backstage/types';
+import { promises as fs } from 'fs';
+import { JWK, exportJWK, importPKCS8, importSPKI } from 'jose';
+import { KeyLike } from 'jose/dist/types/types';
+import { KeyPayload } from './types';
+import { PluginKeySource } from './types';
+
+export type KeyPair = {
+  publicKey: JWK;
+  privateKey: JWK;
+  keyId: string;
+};
+
+export type StaticKeyConfig = {
+  publicKeyFile: string;
+  privateKeyFile: string;
+  keyId: string;
+  algorithm: string;
+};
+
+const DEFAULT_ALGORITHM = 'ES256';
+
+const SECONDS_IN_MS = 1000;
+
+/**
+ * Key source that loads predefined public/private key pairs from disk.
+ *
+ * The private key should be represented using the PKCS#8 format,
+ * while the public key should be in the SPKI format.
+ *
+ * @remarks
+ *
+ * You can generate a public and private key pair, using
+ * openssl:
+ *
+ * Generate a private key using the ES256 algorithm
+ * ```sh
+ * openssl ecparam -name prime256v1 -genkey -out private.ec.key
+ * ```
+ * Convert it to PKCS#8 format
+ * ```sh
+ * openssl pkcs8 -topk8 -inform PEM -outform PEM -nocrypt -in private.ec.key -out private.key
+ * ```
+ * Extract the public key
+ * ```sh
+ * openssl ec -inform PEM -outform PEM -pubout -in private.key -out public.key
+ * ```
+ *
+ * Provide the paths to private.key and public.key as the respective
+ * private and public key paths in the `create` method.
+ */
+export class StaticConfigPluginKeySource implements PluginKeySource {
+  private constructor(
+    private readonly keyPairs: KeyPair[],
+    private readonly keyDurationSeconds: number,
+  ) {}
+
+  public static async create(options: {
+    sourceConfig: Config;
+    keyDuration: HumanDuration;
+  }): Promise<PluginKeySource> {
+    const keyConfigs = options.sourceConfig
+      .getConfigArray('static.keys')
+      .map(c => {
+        const staticKeyConfig: StaticKeyConfig = {
+          publicKeyFile: c.getString('publicKeyFile'),
+          privateKeyFile: c.getString('privateKeyFile'),
```

Comment:
Commented on the config schema already too, but perhaps it makes sense to `getOptionalString` when `index > 0`?